proxy/grpcproxy/watcher.go

```diff
@@ -42,12 +42,13 @@ func (w *watcher) send(wr clientv3.WatchResponse) {
 
 	events := make([]*mvccpb.Event, 0, len(wr.Events))
 
+	var lastRev int64
 	for i := range wr.Events {
 		ev := (*mvccpb.Event)(wr.Events[i])
 		if ev.Kv.ModRevision <= w.rev {
 			continue
 		} else {
-			w.rev = ev.Kv.ModRevision
```

Comment:
txn can have multiple events with the same rev. if we directly update here, we would skip the following events in the same txn.